src/promptflow-devkit/promptflow/_sdk/_utilities/general_utils.py

```diff
@@ -975,6 +974,30 @@ def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code:
         code = Path(code)
         if not code.exists():
             raise UserErrorException(f"Code path {code.as_posix()} does not exist.")
+    return entry, code
+
+
+def create_temp_flex_flow_yaml_without_delete(entry: Union[str, PathLike, Callable], code: Path = None):
+    """
+    Generate a flex flow yaml in temp folder and won't delete it. In flex yaml, the code path points to the original
+    flow folder.
+    """
+
+    from promptflow._sdk._utilities.signature_utils import update_signatures
+    from promptflow._utils.flow_utils import dump_flow_dag_according_to_content
+
+    logger.info("Create temporary entry for flex flow.")
+    entry, code = resolve_entry_and_code(entry, code)
+    flow_dag = {"entry": entry, "code": str(code.resolve())}
+    temp_dir = tempfile.mkdtemp()
+    update_signatures(code=code, data=flow_dag)
+    flow_yaml_path = dump_flow_dag_according_to_content(flow_dag=flow_dag, flow_path=Path(temp_dir))
+    return flow_yaml_path
+
+
+def create_temp_flex_flow_yaml_core(entry: Union[str, PathLike, Callable], code: Path = None):
```

Comment:
May direct change whole `create_temp_flex_flow_xxx`'s behavior to genenerate in tmp folder.

Reply:
updated